HerPublicWebsite.BusinessLogic/Services/QuestionFlowService.cs

```diff
@@ -78,6 +80,11 @@ private QuestionFlowStep ManualAddressBackDestination()
         {
             return QuestionFlowStep.Address;
         }
+        
+        private QuestionFlowStep GasBoilerBackDestination()
+        {
+            return QuestionFlowStep.Address;
```

Comment:
I think this needs to be a bit cleverer. If the user has manually entered their address we don't want to look like we've thrown that away by going all the way back to the address page.
In this case we can probably look at the UPRN - if we found an address on OS places then we should have a UPRN, if not then the user must have entered their address manually.